src/test/java/uk/gov/hmcts/jsonstore/JsonFileStoreWithInheritanceTest.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.jsonstore;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.junit.Test;
+import uk.gov.hmcts.befta.data.HttpTestData;
+
+import java.io.File;
+import java.net.URL;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class JsonFileStoreWithInheritanceTest {
+
+    private JsonFileStoreWithInheritance fileStore;
+
+    private static final String TEST_DATA_RESOURCE_FOLDER = "framework-test-data/json-store-test-data";
+
+    private static final String ID_KEY = "_guid_";
+    private static final String FILE_IN_ROOT_ID = "File-In-Root";
+    private static final String FILE_IN_SUBDIRECTORY_ID = "File-In-Subdirectory";
+
+    @Test
+    public void shouldBuildFileStoreForAllSubdirectoriesSuccessfully() throws Exception {
+        fileStore = new JsonFileStoreWithInheritance(getFileFromResource(TEST_DATA_RESOURCE_FOLDER));
+
+        fileStore.buildObjectStore();
+
+        final JsonNode rootNode = fileStore.getRootNode();
+        assertEquals(2, rootNode.size());
+        assertEquals(FILE_IN_ROOT_ID, rootNode.get(0).get(ID_KEY).textValue());
+        assertEquals(FILE_IN_SUBDIRECTORY_ID, rootNode.get(1).get(ID_KEY).textValue());
+    }
+
+    @Test
+    public void shouldGetObjectWithIdSuccessfully() throws Exception {
+        fileStore = new JsonFileStoreWithInheritance(getFileFromResource(TEST_DATA_RESOURCE_FOLDER));
+
+        final HttpTestData data = fileStore.getObjectWithId(FILE_IN_ROOT_ID, HttpTestData.class);
+
+        assertEquals(FILE_IN_ROOT_ID, data.get_guid_());
```

Comment:
We should check the details of the object. There are lots of fields that should be available inside the object, after applying inheritance on json objects. We should check that al the fields are read into the object and nothing extra.

Reply:
This one is particularly important as 1 single bug in it may cause a lot of trouble of various natures.

Reply:
Using new CommonAssertions class with methods to check the objects that were used elsewhere.